piptools/scripts/compile.py

```diff
@@ -323,6 +324,7 @@ def cli(
                 metadata = project_wheel_metadata(
                     os.path.dirname(os.path.abspath(src_file)),
                     isolated=build_isolation,
+                    runner=default_subprocess_runner,
```

Comment:
By default, `default_subprocess_runner` adds some noise, see the details below. 
<details>

<summary>Details</summary>

```
$ pip-compile pyproject.toml
running egg_info
writing pip_tools.egg-info/PKG-INFO
writing dependency_links to pip_tools.egg-info/dependency_links.txt
writing entry points to pip_tools.egg-info/entry_points.txt
writing requirements to pip_tools.egg-info/requires.txt
writing top-level names to pip_tools.egg-info/top_level.txt
adding license file 'LICENSE'
writing manifest file 'pip_tools.egg-info/SOURCES.txt'
running dist_info
creating /var/folders/l0/lnq1ghps5yqc4vkgszlcz92m0000gp/T/tmpmcmb7ay4/pip_tools.egg-info
writing /var/folders/l0/lnq1ghps5yqc4vkgszlcz92m0000gp/T/tmpmcmb7ay4/pip_tools.egg-info/PKG-INFO
writing dependency_links to /var/folders/l0/lnq1ghps5yqc4vkgszlcz92m0000gp/T/tmpmcmb7ay4/pip_tools.egg-info/dependency_links.txt
writing entry points to /var/folders/l0/lnq1ghps5yqc4vkgszlcz92m0000gp/T/tmpmcmb7ay4/pip_tools.egg-info/entry_points.txt
writing requirements to /var/folders/l0/lnq1ghps5yqc4vkgszlcz92m0000gp/T/tmpmcmb7ay4/pip_tools.egg-info/requires.txt
writing top-level names to /var/folders/l0/lnq1ghps5yqc4vkgszlcz92m0000gp/T/tmpmcmb7ay4/pip_tools.egg-info/top_level.txt
writing manifest file '/var/folders/l0/lnq1ghps5yqc4vkgszlcz92m0000gp/T/tmpmcmb7ay4/pip_tools.egg-info/SOURCES.txt'
adding license file 'LICENSE'
writing manifest file '/var/folders/l0/lnq1ghps5yqc4vkgszlcz92m0000gp/T/tmpmcmb7ay4/pip_tools.egg-info/SOURCES.txt'
creating '/var/folders/l0/lnq1ghps5yqc4vkgszlcz92m0000gp/T/tmpmcmb7ay4/pip_tools-7.3.1.dev17.dist-info'
WARNING: --strip-extras is becoming the default in version 8.0.0. To silence this warning, either use --strip-extras to opt into the new default or use --no-strip-extras to retain the existing behavior.
#
# This file is autogenerated by pip-compile with Python 3.11
# by the following command:
#
#    pip-compile pyproject.toml
#
build==0.10.0
    # via pip-tools (pyproject.toml)
click==8.1.7
    # via pip-tools (pyproject.toml)
packaging==23.1
    # via build
pyproject-hooks==1.0.0
    # via
    #   build
    #   pip-tools (pyproject.toml)
wheel==0.41.1
    # via pip-tools (pyproject.toml)

# The following packages are considered to be unsafe in a requirements file:
# pip
# setuptools

```

</details>


It's unfortunate that pip-tools didn't catch this issue through testing. I suggest:
1. Use `default_subprocess_runner` when pip-compile is in verbose mode, and `quiet_subprocess_runner` otherwise.
2. Add a test to ensure that pip-tools does not produce noise by default with all input formats. This point might be addressed in a follow-up PR.

Reply:
Thanks for the review!
I implemented the first suggestion:
> Use default_subprocess_runner when pip-compile is in verbose mode, and quiet_subprocess_runner otherwise.

In this commit https://github.com/jazzband/pip-tools/pull/1979/commits/e995f16458aad4f1f06ec42755c70e7391f79762

Regarding a second approach, I would stick to a follow-up PR!